src/js/common/helpers/login-helpers.js

```diff
@@ -17,14 +17,30 @@ export function handleLogin() {
 }
 
 export function handleVerify() {
-  this.serverRequest = fetch(`${environment.API_URL}/v0/sessions/new?level=3`, {
+  window.dataLayer.push({
+    event: 'verify-link-clicked',
+  });
+  const urlPromise = fetch(`${environment.API_URL}/v0/sessions/new?level=3`, {
     method: 'GET',
   }).then(response => {
     return response.json();
   }).then(json => {
-    const myVerifyUrl = json.authenticate_via_get;
-    const receiver = window.open(myVerifyUrl, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
-    receiver.focus();
+    return json.authenticate_via_get;
+  });
+
+  // Open the window outside of the promise callback to avoid the browser blocking it.
+  // In case the url fetch fails, we should fix this to behave more gracefully than closing the window.
+  const receiver = window.open('', '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+  urlPromise.then(myVerifyUrl => {
```

Comment:
Yeah, but it's still the value returned by the line `return json.authenticate_via_get;` in the promise chain above. Since that is returned in a promise (`urlPromise`), it's only available asynchronously, so we need to a `.then()` to get a handle on it.

At least that's my understanding from having read up on promises last week, as they were new to me. Just dug up this box analogy which may be helpful (see 1st answer to the question http://stackoverflow.com/questions/22536385/setting-a-variable-to-get-return-from-call-back-function-using-promise)